draft/jumpstarter.rst

```diff
@@ -64,6 +64,115 @@ the other features of Python's ``async/await`` programming model.
+  * Stopping --> The state of the actor while stopping during a restart.
+  * Starting --> The state of the actor while starting during a restart.
+* Restarted --> The state of the actor after it has been restarted.
+
+With these states and sub-states, for both the main state machine and the regular state machine, we provide a clear public API for code to hook into any part of the Actor's Lifecycle. Similar to how, for example, modern asynchronous frontend web frameworks like React and Vue provide hooks into the lifecycle of their components, `transitions`_ provides many different hooks to:
+
+* Have code run before a transition occurs or a state is entered, or conditionally block a transition from happening if certain conditions aren't met.
+* Have code run after a transition occurs (we could use this to, for example, fan out a result right before some hypothetical state ``"task_completed"`` is exited).
+* Do many other things at various granularities and moments. See https://github.com/pytransitions/transitions#callback-execution-order for specific details on the order and timing of when specific callbacks are invoked.
+
+With that base API, Jumpstarter provides a solid foundation and a lot of flexibility to help define self-contained pieces of business logic and facilitate communication between them while maintaining a separation of concerns.
+
+Three abstractions Jumpstarter provides that are addressed in both the ``starting`` and ``stopping`` states are:
+
+1. Dependencies
+2. Resources
+3. Tasks
+
+Dependencies
+------------
+Actors may depend on other actors to run before starting themselves. In some cases, they must depend on another actor if an actor consumes messages from another actor's stream. In `Actor System`_ language, that means that the dependent actor is a parent actor, and the actor consuming messages from the parent actor is the child actor. Just the fact of depending on another actor means that messages will be passed from the parent actor to the child actor (the child actor can also have a way to pass messages back to the parent, but that's out of the scope of this CEP and is something that may be explored as the implementation of Producers and Consumers is more refined).
+
+The proposed public API is as follows:
+
+```
+from jumpstarter import Actor, depends_on
+
+class AccountBalanceActor(Actor):
+  def __init__(self, user_id: int):
+    self.user_id = user_id
+
+class AccountBookkeepingActor(Actor):
+  def __init__(self, user_id: int, account_balance_actor: AccountBalanceActor):
+    self._account_balance_actor = account_balance_actor
+
+  @depends_on
+  def account_balance_actor(self):
+    return account_balance_actor
+```
+
+In this example, the ``AccountBalanceActor`` maintains the balance in a single user ID's account. The ``AccountBookkeepingActor`` is responsible for logging and auditing withdrawals and income, possibly passing these audit logs to another actor responsible for detecting fraud.
```

Comment:
This is still up for discussion. I haven't really given much thought to dependencies.